tests/pipeline/test_events.py

```diff
@@ -1,299 +1,425 @@
+            EventDataSet.next_string: 'string',
+            EventDataSet.next_string_custom_missing: 'string'
+        }
+        cls.previous_value_columns = {
+            EventDataSet.previous_datetime: 'datetime',
+            EventDataSet.previous_event_date: 'event_date',
+            EventDataSet.previous_float: 'float',
+            EventDataSet.previous_int: 'int',
+            EventDataSet.previous_string: 'string',
+            EventDataSet.previous_string_custom_missing: 'string'
+        }
+        cls.loader = cls.make_loader(
+            events=cls.raw_events,
+            next_value_columns=cls.next_value_columns,
+            previous_value_columns=cls.previous_value_columns,
+        )
+        cls.ASSET_FINDER_EQUITY_SIDS = cls.raw_events['sid'].unique()
+        cls.ASSET_FINDER_EQUITY_SYMBOLS = [
+            's' + str(n) for n in cls.ASSET_FINDER_EQUITY_SIDS
+        ]
+        super(EventsLoaderTestCase, cls).init_class_fixtures()
+
+    @classmethod
+    def make_loader(cls, events, next_value_columns, previous_value_columns):
+        # This method exists to be overridden by BlazeEventsLoaderTestCase
+        return EventsLoader(events, next_value_columns, previous_value_columns)
+
+    def test_load_with_trading_calendar(self):
+        engine = SimplePipelineEngine(
+            lambda x: self.loader,
+            self.trading_days,
+            self.asset_finder,
+        )
+
+        results = engine.run_pipeline(
+            Pipeline({c.name: c.latest for c in EventDataSet.columns}),
+            start_date=self.trading_days[0],
+            end_date=self.trading_days[-1],
+        )
 
-class BlazeEventDataSetLoader(BlazeEventsLoader):
-    concrete_loader = EventDataSetLoader
-    _expected_fields = frozenset({ANNOUNCEMENT_FIELD_NAME,
-                                  TS_FIELD_NAME,
-                                  SID_FIELD_NAME})
-
-    def __init__(self,
-                 expr,
-                 dataset=EventDataSet,
-                 **kwargs):
-        super(
-            BlazeEventDataSetLoader, self
-        ).__init__(expr,
-                   dataset=dataset,
-                   **kwargs)
+        for c in EventDataSet.columns:
+            if c in self.next_value_columns:
+                self.check_next_value_results(c, results[c.name].unstack())
+            elif c in self.previous_value_columns:
+                self.check_previous_value_results(c, results[c.name].unstack())
+            else:
+                raise AssertionError("Unexpected column %s." % c)
+
+    def check_previous_value_results(self, column, results):
+        """
+        Check previous value results for a single column.
+        """
+        events = self.raw_events
+        # Remove timezone info from trading days, since the outputs
+        # from pandas won't be tz_localized.
+        dates = self.trading_days.tz_localize(None)
+        for asset, asset_result in results.iterkv():
```

Comment:
Maybe we also want to assert here that results contains all sids that we expect? Otherwise, we're just checking the correctness of assets that we get back from the results, and the test probably shouldn't pass if those are not all there.